app/src/main/java/com/codesoom/assignment/security/UserAuthentication.java

```diff
@@ -5,20 +5,28 @@
 import org.springframework.security.core.GrantedAuthority;
 import org.springframework.security.core.authority.SimpleGrantedAuthority;
 
+import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
 
 public class UserAuthentication extends AbstractAuthenticationToken {
     private final Long userId;
+    private final Role role;
 
-    public UserAuthentication(Long userId, List<Role> roles) {
-        super(authorities(roles));
+    public UserAuthentication(Long userId, Role role) {
+        super(authorities(role));
+        this.role = role;
         this.userId = userId;
     }
 
+    private static List<GrantedAuthority> authorities(Role role) {
+        List<GrantedAuthority> authorities = new ArrayList<>();
```

Comment:
이렇게 하면 어떤 차이점이 있을까요?

```suggestion
        List<GrantedAuthority> authorities = new ArrayList<>(1);
```

Reply:
기존 방법은 개수를 예측할 수 없는 리스트 자료구조를 보는 듯 하고
제안하신 방법은 리스트 자료구조를 가지고 있지만 들어가는 요소가 1개라고 명시적으로 알려주는 느낌입니다.

Reply:
그렇습니다. 그리고 한 가지 더 생각해볼 점이 있습니다. `ArrayList`의 생성자를 읽어보면서 생각해 보세요.

Reply:
크기를 할당하지 않으면 초기 크기가 10인 `ArrayList`가 생성됩니다. 
(실제로는 크기가 0인 배열이 할당되는데 값을 추가하면 크기가 증가하는 로직이 있어 10으로 생성된다는 주석을 적은 것 같습니다.

크기를 할당한다면 해당 크기를 가진 `ArrayList`가 생성됩니다.